MQ2AutoSize.cpp

```diff
@@ -359,269 +536,1087 @@ PLUGIN_API void OnPulse()
+		{
+			ToggleOption("PC", &AS_Config.OptPC);
+		}
+		else if (ci_equals(szNumber, "off") && AS_Config.OptPC)
 		{
-			if (atoi(szCurArg) > 0)
+			ToggleOption("PC", &AS_Config.OptPC);
+			ResetAllByType(PC);
+		}
+		else if (!ci_equals(szNumber, "on") && !ci_equals(szNumber, "off"))
+		{
+			if (!ToggleOption("PC", &AS_Config.OptPC))
 			{
-				AS_Config.ResizeRange = atoi(szCurArg);
-				WriteChatf("\ay%s\aw:: Range set to \ag%d", MODULE_NAME, AS_Config.ResizeRange);
+				ResetAllByType(PC);
 			}
-			else
+		}
+		return;
+	}
+	else if (ci_equals(szCurArg, "npc"))
+	{
+		if (ci_equals(szNumber, "on") && !AS_Config.OptNPC)
+		{
+			ToggleOption("NPC", &AS_Config.OptNPC);
+		}
+		else if (ci_equals(szNumber, "off") && AS_Config.OptNPC)
+		{
+			ToggleOption("NPC", &AS_Config.OptNPC);
+			ResetAllByType(NPC);
+		}
+		else if (!ci_equals(szNumber, "on") && !ci_equals(szNumber, "off"))
+		{
+			if (!ToggleOption("NPC", &AS_Config.OptNPC))
 			{
-				WriteChatf("\ay%s\aw:: Range is \ag%d\ax (was not modified)", MODULE_NAME, AS_Config.ResizeRange);
+				ResetAllByType(NPC);
 			}
-			if (AS_Config.OptAutoSave) SaveINI();
-			return;
 		}
-		else if (!_strnicmp(szCurArg, "size", 5))
+		return;
+	}
+	else if (ci_equals(szCurArg, "everything"))
+	{
+		if (!AS_Config.OptSelf)
+		{
+			DoCommandf("/squelch /autosize self");
+		}
+		if (!AS_Config.OptCorpse)
 		{
-			SetSizeConfig("Default", fNewSize, &AS_Config.SizeDefault);
+			DoCommandf("/squelch /autosize corpse");
 		}
-		else if (!_strnicmp(szCurArg, "sizepc", 7))
+		if (!AS_Config.OptMerc)
 		{
-			SetSizeConfig("PC", fNewSize, &AS_Config.SizePC);
+			DoCommandf("/squelch /autosize mercs");
 		}
-		else if (!_strnicmp(szCurArg, "sizenpc", 8))
+		if (!AS_Config.OptMount)
 		{
-			SetSizeConfig("NPC", fNewSize, &AS_Config.SizeNPC);
+			DoCommandf("/squelch /autosize mounts");
 		}
-		else if (!_strnicmp(szCurArg, "sizepets", 9))
+		if (!AS_Config.OptNPC)
 		{
-			SetSizeConfig("Pet", fNewSize, &AS_Config.SizePet);
+			DoCommandf("/squelch /autosize npc");
 		}
-		else if (!_strnicmp(szCurArg, "sizemercs", 10))
+		if (!AS_Config.OptPC)
 		{
-			SetSizeConfig("Mercs", fNewSize, &AS_Config.SizeMerc);
+			DoCommandf("/squelch /autosize pc");
 		}
-		else if (!_strnicmp(szCurArg, "sizetarget", 11))
+		if (!AS_Config.OptPet)
 		{
-			SetSizeConfig("Target", fNewSize, &AS_Config.SizeTarget);
+			DoCommandf("/squelch /autosize pets");
+		}		
+	}
+	else if (ci_equals(szCurArg, "pets"))
+	{
+		if (ci_equals(szNumber, "on") && !AS_Config.OptPet)
+		{
+			ToggleOption("Pets", &AS_Config.OptPet);
 		}
-		else if (!_strnicmp(szCurArg, "sizemounts", 11))
+		else if (ci_equals(szNumber, "off") && AS_Config.OptPet)
 		{
-			SetSizeConfig("Mounts", fNewSize, &AS_Config.SizeMount);
+			ToggleOption("Pets", &AS_Config.OptPet);
+			ResetAllByType(PET);
 		}
-		else if (!_strnicmp(szCurArg, "sizecorpse", 11))
+		else if (!ci_equals(szNumber, "on") && !ci_equals(szNumber, "off"))
```

Comment:
unnecessary condition. this keeps repeating, i'm not going to comment on any further ones.